### PR TITLE
Parse history and write file

### PR DESCRIPTION
**Pull request by ojung**

### CODE REVIEW

File: src/cmd/root.go [hunk 1]
```diff
@@ -38,7 +41,7 @@ func init() {
 	// Cobra supports persistent flags, which, if defined here,
 	// will be global for your application.
 
-	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.tome{.yaml|.json)")
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.tome{.yaml|.json})")
```
Comment (mweise) [inline src/cmd/root.go:41]: This fixes a bug introduced here: https://github.com/tome-sh/tome/pull/1/commits/47142b9c73620ca67f87636b6831569411352b62 maybe just move it to the other commit?

**Review by mweise — CHANGES_REQUESTED** [review]

File: src/parser.go [hunk 2]
```diff
@@ -17,29 +17,42 @@ type Parser interface {
...
 	file, err := os.Open(filePath)
 	Check(err)
 	defer file.Close()
 
-	buf := make([]byte, batchSizeKb)
 	stat, err := os.Stat(filePath)
 	Check(err)
-	start := stat.Size() - batchSizeKb
+
+	size := stat.Size()
```
Comment (mborst) [inline src/parser.go:80]: Isn't there a better way to express this whole part?
```
start := max(size - batchSize, 0)
buf := make([]byte, min(size, batchSize))
```

File: src/parser_test.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	stat, _ := os.Stat(simpleFile)
+	table := []struct {
+		parser Parser
+		expected string
+	} {
+		{NewZshParser(simpleFile), "md src/test_resources/parser"},
+		{NewZshParserWithBatchSize(simpleFile, int64(stat.Size() / 2)), "md src/test_resources/parser"},
+	}
+
+	for _, tt := range table {
```
Comment (mborst) [inline src/parser_test.go:21]: What's `tt`?

File: README.md [hunk 4]
```diff
@@ -4,6 +4,11 @@ Share shell spells.
```
Comment (mborst) [inline README.md:4]: `Add testing and fix errors` seems kinda arbitrary.
How about one commit like `Add basic test for readSecondToLastLine` and commit(s) before where you fix the bugs?

File: src/parser.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
+package tome
+
+import (
+	"os"
+	"strings"
+)
+
+var batchSizeKb int64 = 512 * 1000
+
+// Parser interface
```
Comment (mborst) [inline src/parser.go]: Why not move the commit where you add linting before this, then you don't need to change it later?

File: src/cmd/root.go [hunk 6]
```diff
@@ -11,6 +11,9 @@ import (
 
 var cfgFile string
 
+var historyFilePathConfigKey = "historyFile"
```
Comment (mborst) [inline src/cmd/root.go]: Should those be constants?
https://golang.org/doc/effective_go.html#constants

**Review by mborst — CHANGES_REQUESTED** [review]

File: src/parser.go (hunk 2 above)
Comment (ojung) [reply src/parser.go:80]: Hahaha, there is no max / min function for integers, _but it's easy to implement your own_ 😂 
https://yourbasic.org/golang/max-min-function/

File: src/cmd/root.go (hunk 6 above)
Comment (ojung) [reply src/cmd/root.go]: https://github.com/tome-sh/tome/pull/1/commits/06b34d6420392eb67ef36cf57f71c91d42bf978a

File: src/parser_test.go (hunk 3 above)
Comment (ojung) [reply src/parser_test.go:21]: test table

File: src/parser.go (hunk 2 above)
Comment (ojung) [reply src/parser.go:80]: https://github.com/tome-sh/tome/pull/1/commits/474f18e6491981065a1fb5edd5cceddab625d0ce

File: README.md (hunk 4 above)
Comment (ojung) [reply README.md:4]: Can we start caring about commit hygiene after the initial bootstraping (this PR)?

**Review by mweise — APPROVED** [review]

**Pull request merged**